library/Freckle/App/Database.hs

```diff
@@ -65,12 +70,63 @@ makePostgresPool = do
   makePostgresPoolWith conf
 
 runDB
-  :: (HasSqlPool app, MonadUnliftIO m, MonadReader app m)
+  :: ( HasSqlPool app
+     , Stats.HasStatsClient app
+     , MonadHandler m
+     , MonadUnliftIO m
+     , MonadReader app m
+     )
   => SqlPersistT m a
   -> m a
 runDB action = do
   pool <- asks getSqlPool
-  runSqlPool action pool
+  mVaultData <- vaultDataFromRequest <$> waiRequest
+  Stats.withGauge Stats.dbConnectionsL $
+    maybe
+      runSqlPool
+      (runSqlPoolXRay "runDB")
+      mVaultData
+      action
+      pool
+
+-- | @'runSqlPool'@ but with XRay tracing
+--
+-- Arguments are ordered to promote composition:
+--
+-- 1. Name before vault data, since Vault data could be @'Maybe'@
+-- 2. Action than pool, to match @'runSqlPool'@
+--
+-- Together, this leads to the nice composition:
+--
+-- @
+-- mVaultData <- getVaultData
+-- maybe 'runSqlPool' ('runSqlPoolXRay' "name") mVaultData action pool
+-- @
+--
+-- N.B. This could make sense in an aws-xray-client-persistent package.
+--
```

Comment:
This note seems kind crufty. No reason to bring it over into the published package, IMO.

```suggestion
```